app/src/main/java/com/example/bottomnav/DoubleDecoder.java

```diff
@@ -2,20 +2,16 @@
 
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
+import java.util.Optional;
 
 public class DoubleDecoder<T> extends PrimitiveDecoder{
 
-    public DoubleDecoder(VariableContents contents) {
-        super(contents);
+    public DoubleDecoder(int size, String name) {
+        super(size, name);
```

Comment:
```suggestion
    public DoubleDecoder(String name) {
        super(8, name);
```

Double (and Float) have standardized sizes, so it shouldn't take size as an argument.

When you're calling a super constructor, you can pass a constant value (or any value), it doesn't necessarily need to come from a containing constructor argument.